tpm2tools/tpm2toolstest/load_random_external_key.go

```diff
@@ -0,0 +1,41 @@
+package tpm2toolstest
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"testing"
+
+	"github.com/google/go-tpm-tools/simulator"
+	"github.com/google/go-tpm/tpm2"
+)
+
+// LoadRandomExternalKey loads a randomly generated external key into the
+// TPM simulator. If any errors occur, calls t.Fatal() on the passed testing.T.
+func LoadRandomExternalKey(t *testing.T, simulator *simulator.Simulator) {
+	t.Helper()
+	pk, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		t.Fatal(err)
+	}
+	public := tpm2.Public{
+		Type:       tpm2.AlgRSA,
+		NameAlg:    tpm2.AlgSHA1,
+		Attributes: tpm2.FlagSign | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth,
```

Comment:
SensitiveDataOrigin doesn't make sense in this context. I'm disappointed if this is not an error.

Reply:
Its not haha. But removed.